packages/patternfly-4/react-charts/src/components/ChartZoomContainer/ChartZoomContainer.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import hoistNonReactStatics from 'hoist-non-react-statics';
+import { VictoryZoomContainer } from 'victory';
+
+class ChartZoomContainer extends VictoryZoomContainer {
+  static propTypes = {
+    ...VictoryZoomContainer.propTypes
+  };
+
+  renderChildren() {
+    const children = React.Children.toArray(this.props.children);
+    return children.map((child) => {
+      // Some victory components expect to control props like domain for
+      // children, some props should be merged rather than overridden
+      const style = _.merge(child.props.style, this.props.style);
+      return React.cloneElement(child, Object.assign({}, child.props, this.props, { style }));
+    });
+  }
+
+  render() {
+    return (
+      <VictoryZoomContainer {...this.props} />);
+  }
+}
+// Note: VictoryZoomContainer.defaultEvents must be hoisted
+hoistNonReactStatics(ChartZoomContainer, VictoryZoomContainer);
+
+export default ChartZoomContainer;
```

Comment:
I don't believe we should extend Victory classes.

"We should not be extending these. It is possible that they could internally remove clasess, etc. A better, and more "React" way, would be to do something like" -- @dmiller9911

```jsx
const ChartArea = (props) => <VictoryArea {...props} />

ChartArea.propTypes = propTypes;
ChartArea.defaultProps = {
   ...VictoryArea.defaultProps,
   theme: ChartTheme.default,
}
```

Another option is to do something like
```jsx
const ChartArea = (props) => <VictoryArea them={ChartTheme.default} {...props} />
```
That would pass the default theme down, and allow the user to overwrite it with a theme prop.

_Originally posted by @dmiller9911 in https://github.com/patternfly/patternfly-react/pull/868_

Reply:
@dlabrecq i don't think you're seeing the issue with the renderChildren. The passing props is not the problem with wrapping the ZoomContainer, its overriding the renderChildren function that provides provides the missing properties I was telling you about, and its the way that Victory recommends wrapping its components. The other components that do not need to manage child props can be wrapped the way you suggest above. Let me know what you think? See this also: https://formidable.com/open-source/victory/guides/custom-components/#wrapping-components

Reply:
The Victory example above shows extending React.Component, but the code is extending VictoryZoomContainer, which is my issue.